### PR TITLE
Add missing documentation comment blocks for fired events

### PR DESCRIPTION
**Pull request by mjauvin**
- most of the examples have a generic "// your code here" in the event handler examples for now, feel free to help with this.

### CODE REVIEW

File: modules/backend/classes/Controller.php [hunk 1]
```diff
@@ -242,8 +242,24 @@ public function run($action = null, $params = [])
...
+         * Example usage:
+         *
+         *     Event::listen('backend.page.beforeDisplay', function ((\Backend\Classes\Controller) $backendController, (string) $action, (array) $param) {
+         *         trace_log('redirect all backend pages to google');
+         *         return \Redirect::to('https://google.com');
+         *     });
+         *
+         * Or
+         *
+         *     $backendController->bindEvent('page.beforeDisplay', function ((string) $action, (array) $param) {
```
Comment (LukeTowers) [inline modules/backend/classes/Controller.php]:
```suggestion
         *     $backendController->bindEvent('page.beforeDisplay', function ((string) $action, (array) $params) {
```

File: modules/backend/classes/NavigationManager.php [hunk 2]
```diff
@@ -100,8 +100,18 @@ protected function loadItems()
             $this->registerMenuItems($id, $items);
         }
 
-        /*
-         * Extensibility
+        /**
+         * @event backend.menu.extendItems
+         * Provides an opportunity to add more menu items
```
Comment (LukeTowers) [inline modules/backend/classes/NavigationManager.php]:
```suggestion
         * Provides an opportunity to manipulate the backend navigation
```

File: modules/backend/models/User.php [hunk 3]
```diff
@@ -136,6 +136,17 @@ public function afterCreate()
     public function afterLogin()
     {
         parent::afterLogin();
+        /**
```
Comment (LukeTowers) [inline modules/backend/models/User.php]: extra line of whitespace before the start of the docblock please

File: modules/system/classes/CombineAssets.php [hunk 4]
```diff
@@ -809,10 +817,19 @@ protected function getCacheKey(array $assets)
...
         }
 
-        /*
-         * Extensibility
-         */
         $dataHolder = (object) ['key' => $cacheKey];
+
+        /**
+         * @event cms.combiner.getCacheKey
+         * Provides an opportunity to modify the $dataHolder->key property in place
```
Comment (LukeTowers) [inline modules/system/classes/CombineAssets.php]:
```suggestion
         * Provides an opportunity to modify the asset combiner's cache key
```

File: modules/system/classes/SettingsManager.php [hunk 5]
```diff
@@ -109,8 +109,17 @@ protected function loadItems()
             $this->registerSettingItems($id, $items);
         }
 
-        /*
-         * Extensibility
+        /**
+         * @event system.settings.extendItems
+         * Provides an opportunity to add interact with the system settings manager
```
Comment (LukeTowers) [inline modules/system/classes/SettingsManager.php]:
```suggestion
         * Provides an opportunity to manipulate the system settings manager
```

File: modules/system/traits/ConfigMaker.php [hunk 6]
```diff
@@ -67,8 +67,19 @@ public function makeConfig($configFile = [], $requiredConfig = [])
...
 
-            /*
-             * Extensibility
+            /**
+             * @event system.extendConfigFile
+             * Provides an opportunity to modify config files
+             *
+             * Example usage:
+             *
+             *     Event::listen('system.extendConfigFile', function ((File) $publicFile, (array) $config) {
```
Comment (LukeTowers) [inline modules/system/traits/ConfigMaker.php]:
```suggestion
             *     Event::listen('system.extendConfigFile', function ((string) $path, (array) $config) {
```

Comment (LukeTowers) [reply modules/system/traits/ConfigMaker.php]: Also double check that it is an array please, I think it's actually a stdClass

File: modules/system/traits/ConfigMaker.php [hunk 7]
```diff
@@ -67,8 +67,19 @@ public function makeConfig($configFile = [], $requiredConfig = [])
...
 
-            /*
-             * Extensibility
+            /**
+             * @event system.extendConfigFile
+             * Provides an opportunity to modify config files
+             *
+             * Example usage:
+             *
+             *     Event::listen('system.extendConfigFile', function ((string) $path, (array) $config) {
```
Comment (bennothommo) [inline modules/system/traits/ConfigMaker.php]: @mjauvin Should `$path` be `$publicFile`, as that's what you use in the line below?
```suggestion
             *     Event::listen('system.extendConfigFile', function ((string) $publicFile, (array) $config) {
```

Comment (LukeTowers) [reply modules/system/traits/ConfigMaker.php]: @bennothommo I suggested $path because it's more clear to the end user what that value is

Comment (bennothommo) [reply modules/system/traits/ConfigMaker.php]: @LukeTowers ah fair enough. @mjauvin You'll just need to change the line below to reflect then :)

File: modules/system/classes/SettingsManager.php [hunk 8]
```diff
@@ -109,8 +109,17 @@ protected function loadItems()
...
 
-        /*
-         * Extensibility
+        /**
+         * @event system.settings.extendItems
+         * Provides an opportunity to manipulate the system settings manager
+         *
+         * Example usage:
+         *
+         *     Event::listen(system.settings.extendItems', function ((\System\Classes\SettingsManager) $settingsManager) {
```
Comment (LukeTowers) [inline modules/system/classes/SettingsManager.php]:
```suggestion
         *     Event::listen('system.settings.extendItems', function ((\System\Classes\SettingsManager) $settingsManager) {
```

**Pull request merged**